From 1153fc4aa57e121f99ca512ca8d641cd2c7c90d6 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 23 Jan 2024 06:35:36 +0200
Subject: [PATCH] [SW-171566] common: fix erom agent creation

Allow endianess setting for erom its
Fix eROM FIT name to exclude _arc suffix (as with other FITs)

Change-Id: I33f9c81a1990cae2c4a2b80661b16e1962e37a6f
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/board/habanalabs/common/agent-fw_loader_erom.its b/board/habanalabs/common/agent-fw_loader_erom.its
index 367f86c..f97f771 100644
--- a/board/habanalabs/common/agent-fw_loader_erom.its
+++ b/board/habanalabs/common/agent-fw_loader_erom.its
@@ -14,7 +14,7 @@
 	images {
 		erom {
 			description = "Erom";
-			data = /incbin/("@HL_DEV@-erom.bin.le");
+			data = /incbin/("@HL_DEV@-erom.bin.@ENDIANNESS@");
 			type = "standalone";
 			compression = "none";
 			offset = <0>;
diff --git a/board/habanalabs/common/post-image-agents.sh b/board/habanalabs/common/post-image-agents.sh
index c5d59be..e861c20 100755
--- a/board/habanalabs/common/post-image-agents.sh
+++ b/board/habanalabs/common/post-image-agents.sh
@@ -61,7 +61,7 @@
 function handle_agent_fit {
 	IMAGE_TYPE=$1
 	ITS_PATH=$2
-	AGENT_ITS_FILE=agent-${AGENT_NAME}-${HL_DEV}-${HL_TARGET}_${IMAGE_TYPE}.its
+	AGENT_ITS_FILE=agent-${AGENT_NAME}-${DEV_PREFIX}-${HL_TARGET}_${IMAGE_TYPE}.its
 	AGENT_ITB_FILE=agent-${AGENT_NAME}-fit_${IMAGE_TYPE}.itb
 
 	AGENT_ITS_TEMPLATE=${ITS_PATH}/agent-${AGENT_NAME}_${IMAGE_TYPE}.its
@@ -70,7 +70,7 @@
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${AGENT_ITS_FILE} -k ${KEYS_DIR} ${AGENT_ITB_FILE}
 
 	# KMD like this to be 4B aligned in size
-	dd status=none conv=sync bs=4 if=${AGENT_ITB_FILE} > ${HL_DEV}-${AGENT_ITB_FILE}
+	dd status=none conv=sync bs=4 if=${AGENT_ITB_FILE} > ${DEV_PREFIX}-${AGENT_ITB_FILE}
 
 	rm -f ${AGENT_ITB_FILE}
 }
